addons/io/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/PollTaskImpl.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.transport.modbus;
```

Comment:
move to internal package

Reply:
I wanted it to be public, usable by modbus bindings (also other than the "generic" org.openhab.binding.modbus).

It is an concrete implementation (specialization when writing "registers") for `ModbusWriteRequestBlueprint` interface, which is referenced in the public `ModbusManager` API.

Makes sense?

Reply:
So do you expect anybody else to implement the PollTask interface at all? If not, there isn't much sense in having the interface and rather rename the PollTaskImpl to PollTask. Usually, the separation is only done in order to keep the implementation hidden.

Reply:
There might cases like that or might not be :) It's way too early to comment since this will be just the beginning of the modbus based bindings!

If you don't mind I would prefer not exposing classes and go with the interfaces instead.

Reply:
But you are exposing the classes :-) So if it is too early for an answer, better keep more things private and only expose it (and thus make it to a public API that you cannot easily change anymore) once it is required.

Reply:
True true :) The usual implementation is exposed (analogy: `ArrayList`) but all the APIs are with interfaces to allow extending (analogy: `List`).

If you feel strongly about this, I can move the implementations as internal to the `org.openhab.binding.modbus`, but that would mean that "simple" modbus device bindings (hopefully we will soo those popping up) would have to come up with the concrete implementations (some of these require correct implementation equals/hash which might be new to some devs)...

Somehow it feels right to me to have the typical implementations available, but keeping the API agnostic and working with the interfaces only.

Reply:
Ok, looking at the analogy with the ArrayList, I would then at least suggest a renaming - "Impl" does suggest to be THE implementation and not one for a special (or rather typical) case. How about calling them `StandardPollTask` or `BasicPollTask` or anything along those lines?

Reply:
Makes sense! I'll do the rename(s).